test/test_helper.rb

```diff
@@ -30,10 +30,82 @@ class ApplicationController < ActionController::Base; end
 
 InheritedResources::Routes = ActionDispatch::Routing::RouteSet.new
 InheritedResources::Routes.draw do
-  get ':controller(/:action(/:id))'
-  get ':controller(/:action)'
-  resources 'posts'
-  root :to => 'posts#index'
+  # aliases_test.rb
+  resources :students
```

Comment:
I see why we need to do this to avoid all the warnings, and it was such a great job to track which route is needed by each test. :+1: :+1:

It'd be great if each test using each route could define & undefine the route on the fly. It would make the tests easier to follow, without having the make all these notes about which test uses which route (comments that definitely make sense in the current situation). But if we can't think of something now, we can leave that improvement for another time.

Reply:
Hah, so you planned to [do this](https://github.com/activeadmin/inherited_resources/pull/459#issuecomment-434509742) already? :smile: :+1:

Reply:
Thanks! Some of the tests can be hard to follow and have some crazy resource setups. 😳 I thought I'd never figure them out as I'm not too familiar with singular resources but eventually I got it all to work. After I had worked my way through several test files I realized that as I added each resource for each test file and ran the test suite, I had previous resources from other tests enabled. So for each test, I commented out all resources and made sure that each test file I listed, only had what resources that file actually depended on, even if it meant it was duplicated there. This way it would be easier to start moving the resource definitions into each test file and not have them be global.

Haha yes I had planned to do it for awhile I'm just not sure yet how I will. At most I was going to try to make building routes into a simple helper method that can be called on setup and then another to remove/reset all routes on teardown.

As a side note, I tried experimenting by changing the tests to be integration tests but ran into what I think is/was a dependency issue. We might not have the right dependencies to enable that. Not that the tests have to change but it came to mind as we rely on rails-controller-testing.

Reply:
Nice strategy to approach this, and also good idea on how to move each route to its own test! :+1:

Reply:
Resolving this as I dropped all those commits now. I have a commit where I remove all global test routes as they have been inlined in each test case. Thanks!